trieste/acquisition/rule.py

```diff
@@ -37,86 +38,72 @@
     SingleModelAcquisitionBuilder,
     SingleModelGreedyAcquisitionBuilder,
 )
-from .optimizer import (
-    AcquisitionOptimizer,
-    automatic_optimizer_selector,
-    batchify,
-    generate_continuous_optimizer,
-)
+from .optimizer import AcquisitionOptimizer, automatic_optimizer_selector, batchify
 from .sampler import ExactThompsonSampler, RandomFourierFeatureThompsonSampler, ThompsonSampler
 
-S = TypeVar("S")
-""" Unbound type variable. """
+T_co = TypeVar("T_co", covariant=True)
+""" Unbound covariant type variable. """
 
 SP_contra = TypeVar("SP_contra", bound=SearchSpace, contravariant=True)
 """ Contravariant type variable bound to :class:`~trieste.space.SearchSpace`. """
 
 
-class AcquisitionRule(ABC, Generic[S, SP_contra]):
-    """The central component of the acquisition API."""
+class AcquisitionRule(ABC, Generic[T_co, SP_contra]):
+    """
+    The central component of the acquisition API.
+
+    An :class:`AcquisitionRule` can produce any value from the search space for this step, and the
+    historic data and models. This value is typically a set of query points, either as is (see e.g.
```

Comment:
"as is", what does this mean?

Reply:
as a `TensorType` rather wrapped in some context, like a state transition: `State[S, TensorType]` or error: `Result[TensorType]` (of course we don't have that in trieste)

Reply:
i agree it's not very clear. I'll see if I can clarify

Reply:
i've expanded on it. how is it now?